### PR TITLE
Add mvapich

### PR DESCRIPTION
**Pull request by j34ni**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This is to ass MVAPIch support (for `linux-`)

### CODE REVIEW

**Comment by j34ni** [comment]
@conda-forge-admin, please rerender

**Comment by j34ni** [comment]
@conda-forge-admin, please rerender

**Comment by j34ni** [comment]
@hmaarrfk Any idea what is wrong with Travis CI here?

**Comment by hmaarrfk** [comment]
I'm not sure. travis is flaky.

**Comment by j34ni** [comment]
@conda-forge-admin, please rerender

**Comment by j34ni** [comment]
@xylar @hmaarrfk It seems that re-rendering fixed it

**Comment by hmaarrfk** [comment]
Honestly Travis has been so unreliable if you think the tests are flaky use emulation and skip the failing ones.

**Comment by j34ni** [comment]
But all the builds succeeded, what exactly failed?

**Comment by hmaarrfk** [comment]
Again. Travis is flaky

**Comment by j34ni** [comment]
Is there something I can do about it?

**Comment by hmaarrfk** [comment]
You should think of your goals. 

Get this PR working. Then start to explore how to fix Travis in an other. To do that. You might decide to start with a much smaller subset of your build.

**Comment by j34ni** [comment]
I am happy to do what you say, but I cannot see which build has failed: everything is green: 0 errors / 0 warnings

**Comment by hmaarrfk** [comment]
You don’t see this

GH failure

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by j34ni** [comment]
@conda-forge-admin, please rerender

**Comment by j34ni** [comment]
@hmaarrfk I skipped the aarch64 build for mvapich for now

**Comment by hmaarrfk** [comment]
I'm going to unsubscribe and let the maintainers execute their review. Ping me again if my help is needed.

**Comment by j34ni** [comment]
@WardF @dopplershift @groutr @kmuehlbauer @mingwandroid @msarahan @ocefpaf
Hi guys, is one of you available to review this PR?

**Comment by xylar** [comment]
@j34ni, it looks like there are a lot of unused patches still here.  Other than that, I'm okay with your changes.  But I would like others to weigh in on disabling tests that use more than 2 CPUs.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -19,15 +19,19 @@ source:
...
+    - patches/do_not_use_more_than_2_processes_in_tests.patch  # [linux]
     - patches/0001-Fix-issue-2674.patch
     - patches/0001-Switch-custom-Bzip2-cmake-module-to-standard.patch
     # removeme once 4.9.3 is released
     - patches/update_ftp_refs.patch
 
+
 build:
   number: {{ build }}
   skip: true  # [win and vc<14]
```
Comment (xylar) [inline recipe/meta.yaml:31]: Let's remove this for cleanliness while we're adding a new skip.  This version isn't being built anyway.

**Comment by j34ni** [comment]
@xylar The reason for disabling tests with 4 to 16 CPUs on Azure is due to limited resources—only 2 CPUs and 6.759 GiB of memory. This causes test failures not related to the actual implementation but rather to the build environment's limitations.

**Comment by j34ni** [comment]
Any more feedback?

**Comment by xylar** [comment]
@conda-forge/libnetcdf, does one of the maintainers with a deeper knowledge of this package want to weigh in on the proposed changes to the tests?  That's the part that I felt a little unsure about here.

**Comment by j34ni** [comment]
@xylar  If that is such an issue I can always restore the original tests for OpenMPI and MPIch

File: recipe/patches/dummy_run_pnetcdf_tests.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (j34ni) [reply recipe/patches/dummy_run_pnetcdf_tests.sh:1]: OK

File: recipe/patches/dummy_tst_parallel_zlib.c [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#include <stdio.h>
```
[the remark this answers is not in the document]
Comment (j34ni) [reply recipe/patches/dummy_tst_parallel_zlib.c:1]: OK

File: recipe/patches/dummy_tst_parallel_compress.c [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#include <stdio.h>
```
[the remark this answers is not in the document]
Comment (j34ni) [reply recipe/patches/dummy_tst_parallel_compress.c:1]: OK

File: recipe/meta.yaml [hunk 5]
```diff
@@ -19,15 +19,23 @@ source:
...
+    - patches/do_not_use_more_than_2_processes_in_tests.patch  # [linux]
+     {% else %}
+    - patches/do_not_use_16_processes_in_tests.patch  # [linux]
+    {% endif %}
     - patches/0001-Fix-issue-2674.patch
     - patches/0001-Switch-custom-Bzip2-cmake-module-to-standard.patch
     # removeme once 4.9.3 is released
     - patches/update_ftp_refs.patch
 
+
```
[the remark this answers is not in the document]
Comment (j34ni) [reply recipe/meta.yaml:32]: OK

**Comment by j34ni** [comment]
There are issues with about a dozen `aarch64` tests **failing to automatically detect the CPU architecture**, which leads to subpar communication performance and eventually timeout
Have you had any issues with that before?
What type of ARM CPUs does QEMU emulate on Azure DevOps?

**Comment by xylar** [comment]
Unfortunately, I don't know the answers to these questions. Anyone else?

**Comment by ocefpaf** [comment]
> What type of ARM CPUs does QEMU emulate on Azure DevOps?

No idea. Maybe @isuruf knows?

**Comment by isuruf** [comment]
Since MVAPICH is ABI compatible with MPICH, I don't see a reason why downstream libraries need to be rebuilt with MVAPICH.

**Comment by ocefpaf** [comment]
> Since MVAPICH is ABI compatible with MPICH, I don't see a reason why downstream libraries need to be rebuilt with MVAPICH.

Good to know. So we can just install one or the other, right?

**Comment by j34ni** [comment]
Thank you @ocefpaf and @isuruf for pointing out the ABI compatibility between MVAPICH and MPICH.

However, their release schedules differ significantly, for instance, the latest MVAPICH version 3.0 is based on MPICH version 3.4.3 from May 2021, while MPICH is about to release version 4.3.0.

This lag means that if we do not rebuild HDF5 with MVAPICH now, we will always be approximately 3.5 years behind in incorporating the latest features and improvements made to HDF5 and all downstream packages.

**Comment by xylar** [comment]
@j34ni, perhaps that means a bit more explanation is needed why supporting mvaich is worth what is already clearly considerable effort and promises to be much more down the line.

**Comment by j34ni** [comment]
Focusing on `linux-64` build for now, since we want to move on with downstream packages

**Comment by ocefpaf** [comment]
@j34ni I understand your points but I'm no longer "in the know" of HPC so I can suggest a good solution here.

While this PR will increase the maintainer burden it does seems to provide real value. That means we need some sort of consensus from a larger group of maintainers. @conda-forge/core, what do you think? Something similar was already merged in hdf5, right?

**Comment by xylar** [comment]
> Something similar was already merged in hdf5, right?

Yes, along with a couple of fixes:
https://github.com/conda-forge/hdf5-feedstock/pull/235
https://github.com/conda-forge/hdf5-feedstock/pull/247
https://github.com/conda-forge/hdf5-feedstock/pull/248

**Comment by j34ni** [comment]
@conda-forge-admin, please rerender

**Comment by j34ni** [comment]
@ocefpaf Is it OK now?

**Comment by xylar** [comment]
@j34ni, my impression is that we need feedback from a broader group of maintainers. Not so much for this feedstock as in order to decide if we all agree that the extra work for maintainers is worth it.

**Comment by j34ni** [comment]
Fair enough

**Comment by hmaarrfk** [comment]
why did you close and delete your work? sometimes these decisions take time, but that doesn't mean that they don't happen.

This is somewhat the nature of collaborative work.

While you wait, you could use

```
python build-locally.py
```
to build the packages you need and upload them to your channel and test them in the real world!

**Pull request closed without merging**